classes/base_decor.rb

```diff
@@ -0,0 +1,31 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.strip
+    if @nameable.correct_name.length > 10
+      @nameable.correct_name[0..9] # the word is cut off at 10 characters
+    else
+      @nameable.correct_name
+    end
+  end
+end
```

Comment:
- [ ] **[OPTIONAL]** We don't really need this `if-else` conditions. We can save code here by doing this instead:

  ```
  class TrimmerDecorator < Decorator
    def correct_name
      @nameable.correct_name[0..9] # the word is cut off at 10 characters
    end
  end
  ```

- [ ] **[OPTIONAL]** I don't think we need this `#strip` method here. The name trimmer works just fine without it.

  ![mso](https://user-images.githubusercontent.com/36542986/192761653-b3ef455d-4c7d-4ed6-b950-7fb397f7ff54.jpg)